plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -1596,5 +1621,128 @@ private static void registerJdkInternalMiscUnsafeIntrinsics(final CompilationCon
+                return lf.literalOf(0L);
+            }
+            LoadedTypeDefinition ltd = objectType.getDefinition().load();
+            FieldElement field = ltd.findField(fieldName);
+            if (field == null) {
+                ctxt.error(builder.getLocation(), "No such field \"%s\" on class \"%s\"", fieldName, ltd.getVmClass().getName());
+                return lf.literalOf(0L);
+            }
+            return builder.offsetOfField(field);
+        };
+
+        intrinsics.registerIntrinsic(unsafeDesc, "objectFieldOffset", classStringToLong, objectFieldOffset1);
+
+        // atomics
+
+        InstanceIntrinsic getAndAddInt = (builder, instance, target, arguments) -> {
+            Value obj = arguments.get(0);
+            Value offset = arguments.get(1);
+            Value incr = arguments.get(2);
+
+            FieldElement fieldElement;
+            if (offset instanceof OffsetOfField) {
+                fieldElement = ((OffsetOfField) offset).getFieldElement();
+            } else {
+                ctxt.error(builder.getLocation(), "Invalid offset argument to %s", target);
+                return ctxt.getLiteralFactory().literalOf(0);
+            }
+            if (fieldElement.isStatic()) {
+                return builder.getAndAdd(builder.staticField(fieldElement), incr, MemoryAtomicityMode.VOLATILE);
+            } else {
+                return builder.getAndAdd(builder.instanceFieldOf(builder.referenceHandle(obj), fieldElement), incr, MemoryAtomicityMode.VOLATILE);
+            }
+        };
+
+        intrinsics.registerIntrinsic(unsafeDesc, "getAndAddInt", objLongIntToInt, getAndAddInt);
+    }
+
+    private static void registerJdkInternalMiscVMIntrinsics(final CompilationContext ctxt) {
+        Intrinsics intrinsics = Intrinsics.get(ctxt);
+        ClassContext classContext = ctxt.getBootstrapClassContext();
+
+        ClassTypeDescriptor vmDesc = ClassTypeDescriptor.synthesize(classContext, "jdk/internal/misc/VM");
+        ClassTypeDescriptor classDesc = ClassTypeDescriptor.synthesize(classContext, "java/lang/Class");
+
+        MethodDescriptor classToVoid = MethodDescriptor.synthesize(classContext, BaseTypeDescriptor.V, List.of(classDesc));
+        MethodDescriptor emptyToVoid = MethodDescriptor.synthesize(classContext, BaseTypeDescriptor.V, List.of());
+
+        Literal voidLiteral = ctxt.getLiteralFactory().zeroInitializerLiteralOfType(ctxt.getTypeSystem().getVoidType());
+
+        StaticIntrinsic initializeFromArchive = (builder, target, arguments) -> voidLiteral;
+
+        intrinsics.registerIntrinsic(vmDesc, "initializeFromArchive", classToVoid, initializeFromArchive);
+
+        StaticIntrinsic initialize = (builder, target, arguments) -> voidLiteral;
+
+        intrinsics.registerIntrinsic(vmDesc, "initialize", emptyToVoid, initialize);
+    }
+
+    private static void registerJavaNetInet4AddressIntrinsics(final CompilationContext ctxt) {
```

Comment:
Defining the init methods as no-ops?  Is this all we need in the long term, or is it a stub to be filled in later?

Reply:
This is all we need. The init methods of these classes are used by JNI to load up field offsets which we won't be needing since our "native" code can access the fields directly.